test-wasm.html

```diff
@@ -0,0 +1,109 @@
+<script src="wasm/harfbuzzjs.js"></script>
+<script>
+var utf8Decoder = new TextDecoder("utf8");
+function bufferToString(buffer) {
+  return utf8Decoder.decode(buffer.slice(0, buffer.indexOf(0)));
+}
+
+var utf8Encoder = new TextEncoder("utf8");
+
+function hb_tag(tag) {
+  return tag.split('').reduce((x, y) => (x << 8) + y.charCodeAt(0), 0);
+}
+
+Module().then(module => {
+  window.module = module;
+
+  fetch('DejaVuSans.ttf').then(res => res.arrayBuffer()).then(fontBlob => {
+    var fontBuffer = module._malloc(fontBlob.byteLength);
+    module.HEAPU8.set(new Uint8Array(fontBlob), fontBuffer);
+
+    var blob = module._hb_blob_create(fontBuffer, fontBlob.byteLength, 2/*HB_MEMORY_MODE_WRITABLE*/, 0, 0);
+    var face = module._hb_face_create(blob, 0);
+    var font = module._hb_font_create(face);
+
+    var buffer = module._hb_buffer_create();
+    {
+      var text = utf8Encoder.encode("سلام!");
+      var text_ptr = module._malloc(text.byteLength);
+      module.HEAPU8.set(text, text_ptr);
+      module._hb_buffer_add_utf8(buffer, text_ptr, text.byteLength, 0, -1);
```

Comment:
The remaining `-1` need to be `text.byteLength` as well.

Reply:
Done. Thanks